src/frame/sdk/GroupApi.ts

```diff
@@ -84,18 +85,46 @@ export function get(groupID: string) {
     return GroupApiEndpoints.callGroupGetApi(groupID).map(formatDetailedGroupResponse);
 }
 
+function createErrorForInvalidUserList(userKeys: UserKeyListResponseType, userGrants: string[]) {
+    const existingUserIDs = userKeys.result.map(({id}) => id);
+    const missingUsersString = userGrants.filter((userID) => existingUserIDs.indexOf(userID) === -1);
+    return missingUsersString;
+}
+
 /**
  * Create a new group with the provided ID and options
  * @param {string}  groupID Client provided optional group ID
  * @param {string}  groupName Client provide optonal group name
  * @param {boolean} addAsMember Whether the group creator shoudl be added as a member upon create
  */
-export function create(groupID: string, groupName: string, addAsMember: boolean) {
-    return GroupOperations.groupCreate(ApiState.userPublicKey(), ApiState.signingKeys(), addAsMember)
-        .flatMap(({encryptedGroupKey, groupPublicKey, transformKey}) =>
-            GroupApiEndpoints.callGroupCreateApi(groupID, groupPublicKey, encryptedGroupKey, groupName, transformKey)
-        )
-        .map((createdGroup) => formatDetailedGroupResponse(createdGroup) as GroupDetailResponse);
+export function create(groupID: string, groupName: string, addAsMember: boolean, userList?: string[]): Future<SDKError, GroupDetailResponse> {
+    const creator = {id: ApiState.user().id, masterPublicKey: publicKeyToBase64(ApiState.userPublicKey())};
+    const creatorMemberKeys = addAsMember ? [creator] : [];
+    const userListResult =
+        userList !== undefined && userList.length > 0
+            ? UserApiEndpoints.callUserKeyListApi(userList).flatMap((userKeysResponse) => {
+                  if (userKeysResponse.result.length !== userList.length) {
+                      const missingUsersString = createErrorForInvalidUserList(userKeysResponse, userList);
+                      return Future.reject(
+                          new SDKError(
+                              new Error(`Failed to create group due to unknown users in user list. Missing user IDs: [${missingUsersString}]`),
+                              ErrorCodes.GROUP_CREATE_WITH_MEMBERS_FAILURE
+                          )
+                      );
+                  }
+                  return Future.of(userKeysResponse.result.map((user) => ({id: user.id, masterPublicKey: user.userMasterPublicKey})));
+              })
+            : Future.of([]);
+
+    return userListResult
+        .map((userMemberKeys) => [...creatorMemberKeys, ...userMemberKeys])
```

Comment:
I'd remove this `.map` and change the logic in the flatMap below. Since `memberKeys` below will always be an array, you can then just do `memberKeys.concat(creatorMemberKey)` as the 3rd parameter to `groupCreate`. Similar to what I mentioned above, once we have that argument always be an array and not an optional array, this change will fit in nicely there.